sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java

```diff
@@ -231,6 +231,10 @@ public PlacementRule run(
             switch (PlacementUtils.getField(fieldName)) {
                 case HOSTNAME:
                     return new MaxPerHostnameRule(1, taskFilter);
+                case ZONE:
+                    return new MaxPerZoneRule(1, taskFilter);
+                case REGION:
+                    return new MaxPerRegionRule(1, taskFilter);
                 case ATTRIBUTE:
                 default:
```

Comment:
Would it make sense to throw `UnsupportedOperationException` in the `default` case? It looks like that'd ensure things visibly blow up if a fifth field type is added, rather than silently treating it as an attribute